### DISCUSSION TITLE
Time Zones Script

### DISCUSSION THREAD

**Discussion by elijahmanor** [johnlindquist/kit#76] [Share]
![Kapture 2021-03-25 at 07 53 12](https://user-images.githubusercontent.com/86454/112476058-7d534f80-8d3f-11eb-8734-013cfeb9d6cf.gif)

The company I work for has employees all over the world, so I thought it'd be fun to make a Time Zone script in Kit.

Here is the code...

```js
// Menu: Time Zones
// Description: List of world times
// Author: Elijah Manor
// Twitter: @elijahmanor

const ct = await npm('countries-and-timezones');
const scriptDB = db('timezone', { recents: [] });
const timeZones = ct.getAllTimezones();

const now = new Date();

const addToRecents = async () => {
  const items = Object.keys(timeZones)
    .map((name) => {
      const country = ct.getCountry(timeZones[name].country);
      if (!country) {
        return null;
      }

      return {
        name: `${name} (${country.name})`,
        value: name,
        description: `${now.toLocaleTimeString('en-US', {
          hour: '2-digit',
          timeZone: name,
          minute: '2-digit',
          hour12: true,
          weekday: 'short',
          month: 'short',
          day: 'numeric',
        })} (${timeZones[name].utcOffsetStr})`,
      };
    })
    .filter((x) => x);

  const timeZone = await arg('Select to add to recents...', items);
  const recents = scriptDB.get("recents").value();
  recents.push(items.find((t) => t.value === timeZone));
  scriptDB.set("recents", recents).write();
};

const removeFromRecents = async () => {
  let recents = scriptDB.get('recents').value();
  const timeZone = await arg('Select to remove from recents...', recents);
  recents = recents.filter((r) => r.value !== timeZone);
  scriptDB.set('recents', recents).write();
};

onTab('Search', addToRecents);

onTab('Recent', removeFromRecents);
```

**Comment by johnlindquist** [comment]
[Click to install Elijah's script](https://scriptkit.app/api/new?name=time-zones&url=https://gist.githubusercontent.com/johnlindquist/014389b94f70764b9ee51a8613187ae2/raw/1fc9ab6ce6f61f0086cc4ee463e543e7c142f2c1/time-zones)

Oh, love this! I tried doing this with an API before, but hit some snags with their restrictions. Glad that you figured it out because I'm totally using this (already installed) :D

Related: I've been thinking about hooking up the [Google People API][https://developers.google.com/people] so I could query my contacts and check the timezone offset just like you're doing here. I can definitely see a good use case for "Search person, display time difference"

**Reply by elijahmanor** [reply]
Oh that would be nice to hook up to the Google People API. Great idea
